spec/analytics_dept_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'spec_helper'
+
+RSpec.describe AnalyticsDept do
+  before(:each) do
+    @analytics_dept = AnalyticsDept.new
+  end
+  describe '#initialize' do
+    it 'can initialize' do
+      expect(@analytics_dept).to be_an_instance_of(AnalyticsDept)
+    end
+
+    it 'starts with no data' do
+      expect(@analytics_dept.current_data).to eq(nil)
+    end
+  end
+
+  describe '#get_data' do
+    it 'can get data and assign it as current data' do
+      @analytics_dept.get_data
+      expect(@analytics_dept.current_data).to be_an_instance_of(Array)
+      @analytics_dept.current_data.each do |vehicle|
+        expect(vehicle).to be_an_instance_of(Vehicle)
+      end
+    end
+  end
+
+  describe '#find_popular' do
+    it 'can find the most popular make' do
+      @analytics_dept.get_data
+      expect(@analytics_dept.find_popular(:make)).to be_an_instance_of(String)
+    end
+
+    it 'can find the most popular model' do
+      @analytics_dept.get_data
+      expect(@analytics_dept.find_popular(:model)).to be_an_instance_of(String)
+    end
+
+    it 'will not run if there is no data' do
+      expect(@analytics_dept.find_popular(:make)).to be false
+      expect(@analytics_dept.find_popular(:model)).to be false
+    end
```

Comment:
This is a good test. As a thought experiment, how could you use mocks and stubs to make the test more realistic? 
For example, what if this test included `@analytics_dept.get_data`, and it was somehow set up to return nil?